lldb/source/Symbol/Block.cpp

```diff
@@ -399,11 +392,9 @@ size_t Block::MemorySize() const {
   return mem_size;
 }
 
-void Block::AddChild(const BlockSP &child_block_sp) {
-  if (child_block_sp) {
-    child_block_sp->SetParentScope(this);
-    m_children.push_back(child_block_sp);
-  }
+BlockSP Block::CreateChild(user_id_t uid) {
+  m_children.push_back(std::shared_ptr<Block>(new Block(uid, *this)));
```

Comment:
make_shared won't work because that constructor is private. I think protecting that one is important as it ensures the integrity of the parent-child links.

That said, the tagged constructor approach does make it possible to use `make_shared`, so this could be another argument in favour of it.